homer0/projext
# Automatically detect the build engine

### What does this PR do?

Until now, the build engine by default was webpack, but since the plugin for Rollup is soon to be released, I changed the way the project configuration service handles that:

The service now has a list of _"known build engine"_, webpack and Rollup; when creating the configuration it will check if a plugin with a name matching any of them was registered (`projext-plugin-[webpack|rollup]`) and use it as the default value.

In order to do all the things described above, the following changes have been done:

1. `engine` default value was set `null` on the targets' templates. It get's updated when the configuration is created and if a plugin for a build engine was registered.
2. If you didn't install any build engine and try to build a browser target or a Node target that requires bundling, you'll get and error.

This is breaking because engine is now a nullable property, so even if a build engine plugin is not installed, if other service/plugin reads a target settings, it will now get a value that can, sometimes, be `null`.

### How should it be tested manually?

Create a project without installing the webpack plugin and try to build/run a browser target. You should get an error.

Install the webpack plugin on the project and try again, everything should work as always.

And of course...

### How should it be tested manually?

```bash
npm test
# or
yarn test
```
